### PR TITLE
Minor User & UserGroup properties fixes

### PR DESCRIPTION
**Pull request by andrerom**
Some cleanup on user properties in regards to how internal properties are handled.

### CODE REVIEW

File: eZ/Publish/Core/Repository/Values/User/User.php [hunk 1]
```diff
@@ -140,18 +140,19 @@ public function __get($property)
...
                 return $this->getVersionInfo();
 
             case 'fields':
                 return $this->getFields();
+
+            case 'content':
+                @trigger_error(
+                    sprintf('%s is and internal property. User itself exposes everything needed.', $property),
+                    E_USER_DEPRECATED
+                );
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Values/User/User.php]: Nitpick: as this `trigger_error` does not stop execution, after this line there should be a `break;` (easy to forget when adding next `case` in the future). However this does not change anything right now, so only nitpicking :)

File: eZ/Publish/Core/Repository/Tests/Values/User/UserTest.php [hunk 2]
```diff
@@ -55,6 +55,31 @@ public function testMissingProperty()
...
+        $object = new User();
+        $properties = $object->attributes();
+        self::assertNotContains('internalFields', $properties, 'Internal property found ');
+        self::assertContains('id', $properties, 'Property not found ');
+        self::assertContains('fields', $properties, 'Property not found ');
+        self::assertContains('versionInfo', $properties, 'Property not found ');
+        self::assertContains('contentInfo', $properties, 'Property not found ');
+
+        // check for duplicates and double check existence of property
+        $propertiesHash = array();
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Tests/Values/User/UserTest.php]: Nitpick: I'm still in love with short array syntax.

File: eZ/Publish/Core/Repository/Values/User/UserGroup.php [hunk 3]
```diff
@@ -101,7 +101,7 @@ public function getField($fieldDefIdentifier, $languageCode = null)
      *
      * @return array
      */
-    protected function getProperties($dynamicProperties = array('id', 'contentInfo'))
+    protected function getProperties($dynamicProperties = array('id', 'contentInfo', 'versionInfo', 'fields'))
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Values/User/UserGroup.php]: Nitpick: same request here for short array syntax if possible.

File: eZ/Publish/Core/Repository/Values/User/UserGroup.php [hunk 4]
```diff
@@ -120,18 +120,19 @@ public function __get($property)
...
                 return $this->getVersionInfo();
 
             case 'fields':
                 return $this->getFields();
+
+            case 'content':
+                @trigger_error(
+                    sprintf('%s is and internal property. UserGroup itself exposes everything needed.', $property),
+                    E_USER_DEPRECATED
+                );
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Values/User/UserGroup.php]: Nitpick: the same as above, about missing `break;`

File: eZ/Publish/Core/Repository/Values/User/User.php (hunk 1 above)
Comment (andrerom) [inline eZ/Publish/Core/Repository/Values/User/User.php]: It's on purpose, I'll add a comment about that ;)

**Pull request merged**